storage/remote/wal_watcher.go

```diff
@@ -78,6 +78,7 @@ var (
 		},
 		[]string{queue},
 	)
+	liveReaderMetrics = wal.NewLiveReaderMetrics(prometheus.DefaultRegisterer)
```

Comment:
I'd love to see us refactor this eventually to not use global :) not a blocker here though

Reply:
Would also love to see that, I think the struct has to be exported to enable that and then we would have to pass it down from somewhere pretty high. Seemed like a more invasive change than making it global for now.

Reply:
Definitely, just evangelizing this, this pattern is pretty spread around the codebase right now, so definitely something to solve here :)

Reply:
I can do this as part of prometheus/tsdb#606